tools/uuid-corrector/src/uuid_corrector/core.clj

```diff
@@ -0,0 +1,137 @@
+
+(def ^:private cli-options
+  [["-h" "--host HOST" "Database host name"
+    :default "localhost"
+    :validate [(complement string/blank?) "Must not be blank."]]
+   ["-p" "--port PORT" "Database port number"
+    :default 5432
+    :parse-fn #(Integer/parseInt %)
+    :validate [#(< 0 % 65536) "Must be an integer between 0 and 65536."]]
+   ["-U" "--user USER" "Database username"
+    :default "de"
+    :validate [(complement string/blank?) "Must not be blank"]]
+   ["-d" "--de-database DATABASE" "DE database name"
+    :default "de"
+    :validate [(complement string/blank?) "Must not be blank"]]
+   ["-m" "--metadata-database DATABASE" "Metadata database name"
+    :default "metadata"
+    :validate [(complement string/blank?) "Must not be blank"]]
+   ["-?" "--help" "Display this help message"]])
+
+(defn- bad-usage
+  [summary errors]
+  (println "Invalid Usage:")
+  (dorun (map (comp println (partial str "\t")) errors))
+  (println)
+  (println summary)
+  (System/exit 1))
+
+(defn- show-help
+  [summary]
+  (println summary)
+  (System/exit 0))
+
+(defn- prompt-for-password
+  "Prompts the user for a password."
+  [user]
+  (print user "password: ")
+  (flush)
+  (.. System console readPassword))
+
+(defn- get-password
+  "Attempts to obtain the database password from the user's .pgpass file.  If
+  the password can't be obtained from .pgpass, prompts the user for the
+  password"
+  [host port database user]
+  (let [password (pgpass/get-password host port database user)]
+    (if (nil? password)
+      (if (nil? (System/console))
+        (binding [*out* *err*]
+          (println "No password supplied.")
+          (System/exit 1))
+        (prompt-for-password user))
+      password)))
+
+(defn- build-spec
+  [host port db user password]
+  {:subprotocol "postgresql"
+   :subname     (str "//" host ":" port "/" db)
+   :user        user
+   :password    password})
+
+(def uuid-columns [:created_by :modified_by])
+(def tables [:templates :attributes])
+(def table-columns (partition 2 (flatten (map (fn [table] (map #(list table %) uuid-columns)) tables))))
+
+(defn- update-uuids
+  [table column uuid username]
+  (-> (update table)
+      (sset {column username})
+      (where [:= column (sql/call :cast uuid :varchar)])))
+
+(defn- get-usernames
+  [uuids]
+  (println uuids)
+  (-> (select :id (sql/call :regexp_replace :username "@iplantcollaborative.org$" ""))
+      (from :users)
+      (where [:in (sql/call :cast :id :varchar) uuids])))
+
+(defn- get-uuids
+  []
+  {:union (map (fn [[table column]] (-> (select column) (from table))) table-columns)})
+
+(defn- migrate-table
+  [table meta-trans uuid-mappings]
+  (println table)
+  (println uuid-mappings)
+  (dorun (map
+    (fn [column] (println column) (dorun (map
+      (fn [[uuid username]]
+        (println (sql/format (update-uuids table column uuid username)))
+        (jdbc/execute! meta-trans (sql/format (update-uuids table column uuid username))))
```

Comment:
I don't use it often, but `for` (http://clojuredocs.org/clojure.core/for) would be useful here. It would help to get rid of the nested `(dorun (map ...))` calls. `for` produces a lazy sequence, so a `dorun` would still be necessary, but at least you'd only need one.